sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -585,34 +589,6 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
         }
 
       case p: LogicalPlan if p.resolved => p
-
-      case p @ CreateTableAsSelect(db, tableName, child, allowExisting, None) =>
```

Comment:
`CreateTableAsSelect` was designed to support CTAS, not only for HiveQL, but also for the other SQL dialect. See: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala#L128, the type of `desc` is `desc:Option[T]`
In HiveQL case, the `T` should be `ASTNode`, and `desc` will never be `None`, we can always extract the `SerDe` information from it via Hive `SemanticAnalyzer`, even if no storage spec specified in the DDL SQL, as Hive `SemanticAnalyzer` will give the default `SerDe` for that case, So seems `hasStorageSpec` will be false only if the table already existed?

The change @viirya made seems reasonable for me here, the only concern is what if `hive.convertCTAS` is false and no storage spec specified in DDL SQL? How to extract the default serde (as the `desc` is None)?